keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -35,38 +42,51 @@ class MemoryCloudDriverCache(
   private val securityGroupSummariesByIdOrName = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, MINUTES)
-    .build<String, SecurityGroupSummary>()
+    .buildAsync<String, SecurityGroupSummary>()
 
   private val networks = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Network>()
+    .buildAsync<String, Network>()
 
   private val availabilityZones = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Set<String>>()
+    .buildAsync<String, Set<String>>()
 
   private val credentials = Caffeine.newBuilder()
     .maximumSize(100)
     .expireAfterWrite(1, HOURS)
-    .build<String, Credential>()
+    .buildAsync<String, Credential>()
 
   private val subnetsById = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Subnet>()
+    .buildAsync<String, Subnet>()
 
   private val subnetsByPurpose = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Subnet>()
+    .buildAsync<String, Subnet>()
 
   override fun credentialBy(name: String): Credential =
     credentials.getOrNotFound(name, "Credentials with name $name not found") {
       cloudDriver.getCredential(name, DEFAULT_SERVICE_ACCOUNT)
     }
 
+  /**
+   * Convert a value to a [CompletableFuture] so it can be passed to [AsyncCache.put]
+   */
+  private fun <T> futurify(x: T?): CompletableFuture<T?> = CoroutineScope(EmptyCoroutineContext).future { x }
+
+  /**
+   * Convert a suspending function to a [BiFunction] that [AsyncCache.get] expects as its second argument
+   *
+   * Uses [Dispatchers.IO] because it assumes that [block] makes I/O calls
+   */
+  private fun <T> asyncify(block: suspend CoroutineScope.() -> T?): BiFunction<String, Executor, CompletableFuture<T?>> =
+    BiFunction { _: String, _: Executor -> CoroutineScope(Dispatchers.IO).future(block = block) }
```

Comment:
What's the intent of specifying the types here if you don't use the variables?

Reply:
They don't really need to be there. They are a remnant of earlier struggles I was having figuring out how to convert from a kotlin closure to a BiFunction. I'll zap them.

Reply:
Zapped in 83982e61